### DISCUSSION TITLE
Leak from magics usage

### DISCUSSION THREAD

**Discussion by byronanderson** [alpinejs/alpine#3289] [7. Bugs (archived)]
High frequency evaluations of alpine strings that use magics (requestAnimationFrame usage in our case) exacerbate a leak we have found.

What we observe is that _x_cleanups will grow by 1 every time you use a magic property in an evaluated string, which will eventually cause our web page to crash running out of memory.

Here is a minimal reproduction of the issue: https://codesandbox.io/s/alpine-memory-broken-mlip56?file=/index.html

Resize the preview window on that page and you should see that _x_cleanups for the div with id "leaky" grows.

[This other issue](https://github.com/alpinejs/alpine/issues/2140) appears to be unrelated since incorporating the patch from his branch doesn't alleviate the leak we are experiencing.</div>

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
